### PR TITLE
Add Workflow run and dispatch events

### LINKED ISSUE #1598: Add support for workflow_dispatch & workflow_run webhook events
Github has added two new webhook events.
1. [workflow_dispatch](https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#workflow_dispatch)
2. [workflow_run](https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#workflow_run)

Those are not implemented currently. I can pick it up if it's ok.

### PR DESCRIPTION
**Pull request by nicolasmanic**
Add support for [workflow_dispatch] & [workflow_run] webhook events.

[workflow_dispatch]: https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#workflow_dispatch
[workflow_run]: https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#workflow_run

Fixes #1598

### CODE REVIEW

**Comment by nicolasmanic** [comment]
@googlebot I signed it!

File: github/event.go [hunk 1]
```diff
@@ -118,6 +118,10 @@ func (e *Event) ParsePayload() (payload interface{}, err error) {
 		payload = &UserEvent{}
 	case "WatchEvent":
 		payload = &WatchEvent{}
+	case "WorkflowRunEvent":
+		payload = &WorkflowRunEvent{}
+	case "WorkflowDispatchEvent":
```
Comment (gmlewis) [inline github/event.go:123]: Please move this one above to keep these in sorted order.

File: github/messages_test.go [hunk 2]
```diff
@@ -359,6 +359,13 @@ func TestParseWebHook(t *testing.T) {
 			payload:     &RepositoryDispatchEvent{},
 			messageType: "repository_dispatch",
 		},
+		{
+			payload:     &WorkflowDispatchEvent{},
+			messageType: "workflow_dispatch",
+		}, {
```
Comment (gmlewis) [inline github/messages_test.go:365]: Was this file run through `gofmt` ?

I'm really surprised it didn't catch this.

Could you please change this to make it more consistent with the rest of the repo?

```
},
{
```

Comment (nicolasmanic) [reply github/messages_test.go]: it was, but now it should be fine.

**Comment by nicolasmanic** [comment]
Thanks @nightlark @gmlewis for the reviews.

**Pull request merged**